vpr/src/draw/buttons.cpp

```diff
@@ -53,6 +91,36 @@ void button_for_toggle_nets() {
                              toggle_nets_widget);
 }
 
+void button_for_net_max_fanout() {
+    GObject* main_window = application.get_object(application.get_main_window_id().c_str());
+    GObject* main_window_grid = application.get_object("InnerGrid");
+
+    //find maximum fanout
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& clb_nlist = cluster_ctx.clb_nlist;
+    size_t max_fanout = 0;
+    for (ClusterNetId net_id : clb_nlist.nets())
+        max_fanout = std::max(max_fanout, clb_nlist.net_sinks(net_id).size());
+
+    //spin box for net_max_fanout, set the range and increment step
+    GtkWidget* net_max_fanout_widget = gtk_spin_button_new_with_range(0, std::numeric_limits<int>::max(), 1.);
```

Comment:
Shouldn't the high-end of this range be `max_fanout`, rather than std::numeric_limits<int>::max()?

Currently, if the user enters a large number (e.g. `999999999999`which is much greater than max_fanout), it is replaced with `2147483647` (int max). Would be better to replace it with the value of `max_fanout`.